registry/registry.go

```diff
@@ -737,24 +811,299 @@ func downloadVersion(opts *VersionOptions) (ver *Version, attachments []*kivik.A
 		} else if tryCount <= 3 {
 			continue
 		} else {
-			return
+			return nil, nil, err
 		}
 	}
 
+	// Reader for filesize
 	counter := &Counter{}
 	var reader io.Reader = buf
 	reader = io.TeeReader(reader, counter)
 
-	var packVersion string
+	// Reader for tarball content
+	var fileContent = new(bytes.Buffer)
+	reader = io.TeeReader(reader, fileContent)
+
+	// Reading the tarball content
+	tarball, err := ReadTarballVersion(reader, contentType, url)
+	if err != nil {
+		return nil, nil, err
+	}
+	tarball.ContentType = contentType
+
+	return tarball, counter, nil
+}
+
+func downloadVersion(opts *VersionOptions) (ver *Version, attachments []*kivik.Attachment, err error) {
+	var tarPrefix string
+	url := opts.URL
+
+	tarball, counter, errd := downloadTarball(opts, url)
+	if err != nil {
+		err = multierror.Append(err, errd)
```

Comment:
If we have an error here, I don't see the purpose to continue the execution of this function. It will panic on reading `tarball.Manifest` 3 lines after that.